sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1059,9 +1058,18 @@ def test_valid_brute_metric_for_auto_algorithm():
     require_params = ['mahalanobis', 'wminkowski', 'seuclidean']
     for metric in VALID_METRICS['brute']:
         if metric != 'precomputed' and metric not in require_params:
-            nn = neighbors.NearestNeighbors(n_neighbors=3, algorithm='auto',
-                                            metric=metric).fit(X)
-            nn.kneighbors(X)
+            if metric == 'haversine':
```

Comment:
This duplicates a lot more than necessary, when really we just want to switch `X if metric != 'haversine' else X_haversine`